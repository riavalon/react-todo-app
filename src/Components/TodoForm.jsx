import React from 'react'
import Button from './Button'
import Input from './Input'

class TodoForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    return (
      <form onSubmit={this.props.addItem}>
        <Input 
          placeholder="Start typing here..." 
          value={this.props.currentItem.text}
          onChange={this.props.handleInput}
        />
        <Button text="Add Task" />
      </form>
    )
  }
}

export default TodoForm