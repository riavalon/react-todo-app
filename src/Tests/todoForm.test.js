import React from 'react';
import renderer from 'react-test-renderer';
import TodoForm from '../Components/TodoForm';

it('renders correctly', () => {
  const tree = renderer.create(
    <TodoForm currentItem={{ text: 'Thing to do', key: 1}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});