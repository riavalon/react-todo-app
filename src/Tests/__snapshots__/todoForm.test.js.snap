// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <input
    className="input"
    placeholder="Start typing here..."
    type="text"
    value="Thing to do"
  />
  <button
    className="button"
    type="submit"
  >
    Add Task
  </button>
</form>
`;
